lib/puppet/provider/package/pkg.rb

```diff
@@ -227,6 +234,10 @@ def install(nofail = false)
     if is[:ensure].to_sym == :absent
       command = 'install'
     else
+      # If the package is up to date, we don't need to do anything
+      if insync?(should)
+        return 0
+      end
```

Comment:
I wouldn't think this is necessary. Puppet's resource harness calls the `insync?` method on this provider, and only if it returns false should the `install` method be called.

Reply:
I noticed that without this, we were still progressing through it.

Reply:
Specifically, I noticed that we were still doing the work in it where the package was already up to date. That is why I added it. Removing it will reduce the positive impact of the fix.

Reply:
After some testing I've removed this check.